include/deal.II/base/mpi.templates.h

```diff
@@ -326,9 +327,10 @@ namespace Utilities
     sum(const Tensor<rank, dim, Number> &local,
         const MPI_Comm &                 mpi_communicator)
     {
-      Tensor<rank, dim, Number> sums;
-      sum(local, mpi_communicator, sums);
-      return sums;
+      auto array = local.to_array();
+      decltype(array) result;
+      sum(ArrayView<const Number>(array), mpi_communicator, ArrayView<Number>(result));
```

Comment:
`sum` is documented as allowing the first and last arguments to be the same. You can avoid the `results` variable altogether.

Reply:
I ended up not changing this function (I do have the patches, but they will be in a followup PR) so I am marking this as resolved.